Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js

```diff
@@ -195,17 +202,8 @@ const ToolbarAndroid = createReactClass({
     }
 
     return <NativeToolbar onSelect={this._onSelect} {...nativeProps} />;
-  },
-
-  _onSelect: function(event) {
-    const position = event.nativeEvent.position;
-    if (position === -1) {
-      this.props.onIconClicked && this.props.onIconClicked();
-    } else {
-      this.props.onActionSelected && this.props.onActionSelected(position);
-    }
-  },
-});
+  }
+}
```

Comment:
To successfully get rid of `NativeMethodMixin`, you need to do this:

```JavaScript
const ToolbarAndroidWithRef = React.forwardRef((props: Props, ref: React.Ref<typeof NativeToolbar>) => {
  return <ToolbarAndroid {...props} forwardedRef={ref} />
});
```

And then inside `ToolbarAndroid`, you need to do this:
```JavaScript
type Props = $ReadOnly<{|
  // ...
  forwardedRef: React.Ref<typeof NativeToolbar>,
|}>

class ToolbarAndroid extends React.Component<Props> {
  // ...
  render() {
    const {
      onIconClicked,
      onActionSelected,
      forwardedRef: ref,
      ...nativeProps,
    } = this.props;    
    // ...

    return <NativeToolbar onSelect={this._onSelect} {...nativeProps} ref={ref}/>
  }
}
```

I think React.forwardRef isn't flow typed yet, so you'll have to find a workaround for that. There are other PRs removing NativeMethodsMixin that accomplish this (e.g: #21885)

Reply:
The bit that needs to be copied for now specifically is the FlowFixMe comment above the forwardref call, to suppress the type errors

Reply:
👍